app/controllers/concerns/foreman/controller/registration_commands.rb

```diff
@@ -5,7 +5,7 @@ module Foreman::Controller::RegistrationCommands
 
   def command
     args_query = "?#{registration_args.to_query}"
-    "curl -sS #{insecure} '#{registration_url(@smart_proxy)}#{args_query if args_query != '?'}' #{command_headers} | bash"
+    "output=$(curl -sS #{insecure} '#{registration_url(@smart_proxy)}#{args_query if args_query != '?'}' #{command_headers}) && echo \"$output\" | bash"
```

Comment:
> I was thinking we could save the curl command output in a temporary file and execute it only in the case of a successful curl exit status.

I would agree. It's better to save curl's output into a temp file, and then redirecting it into bash. This will be much cleaner and less prone to errors due to special symbols.

```suggestion
    "tmp=$(mktemp) && curl -o $tmp -sS #{insecure} '#{registration_url(@smart_proxy)}#{args_query if args_query != '?'}' #{command_headers} && cat #tmp | bash"
```

This should do the trick, but I didn't test it...